lib/handlers.js

```diff
@@ -22,10 +22,42 @@ export const makeHandlers = ({_, config}) => {
         bundleTrigger: 'dependencies.js'
       }, options)
 
-      if ((typeof bundleTrigger === 'string' && req.originalUrl.endsWith(bundleTrigger)) ||
-        (typeof bundleTrigger == 'function' && bundleTrigger(req))) {
-        _.log(`requesting ${req.url} triggering bundling`)
-        _.bundle(config.entries.join(' + ')).then(({source}) => res.end(source))
+      // Request can match with string, function or regex.
+      const matchRequest = (req, match) =>
+        typeof match === 'string' && (req.originalUrl.endsWith(match) || req.originalUrl.endsWith(match + '.map')) ||
+        typeof match === 'function' && match(req) ||
+        typeof match === 'object' && (match instanceof RegExp) && match.exec(req.originalUrl);
+
+      // Simple case: mapping bundle trigger to request.
+      let builder;
+      if (matchRequest(req, bundleTrigger)) {
+        builder = {expression: config.entries.join(' + ')}
+
+      // Didn't match; look for mappings overrides.
+      } else {
+        (config.mappings || []).forEach(mapping => {
+          if (matchRequest(req, mapping.match)) {
+            if (!mapping.builder) {
+              throw new Error('Builder options missing for mapping: ' + mapping.match);
+            }
+            builder = mapping.builder;
+          }
+        });
+      }
+
+      // Is this a mapped builder request?
+      if (builder) {
+        const options = merge(config.builder.options, builder.options);
+        if (req.originalUrl.endsWith('.map')) {
+          _.log(`requesting ${req.url} retrieving external sourcemap`)
+          _.bundle(builder.expression, options).then(({sourceMap}) => res.end(sourceMap))
+        } else {
+          _.log(`requesting ${req.url} triggering bundling`)
+          _.bundle(builder.expression, options).then(({source, sourceMap}) => res.end(
+            options.sourceMaps === 'inline' || !sourceMap
+              ? source
```

Comment:
Not sure how much I'm a fan of building this in here... But it's ok for now. Might rework it in december.